src/PAModel/Schemas/Schema.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerPlatform.Formulas.Tools.Schemas
+{
+    // Define Schema.yaml - strong typing for input and output parameters
+    public class ParameterSchema
+    {
+        // What names here?
+        public enum ParamType
+        {
+            Number,
+            String,
+            Boolean,
+            Date,
+            Time,
+            DateTime,
+            Color,
+            Guid,
+            Record,
+            Table,
+            EntityRecord,
+            EntityTable,
+        }
+
+        public class Parameter
+        {
+            public ParamType Type { get; set; }
+            public bool Required { get; set; }
+
+            // Optional: For entities, specifies the name
+            public string Name { get; set; }
+
+            // Optional: For records, can become recursive
+            public Dictionary<string, Parameter> Fields { get; set; }
+        }
+                
+        // Input parameters (public)
+        public Dictionary<string, Parameter> Parameters { get; set; }
```

Comment:
Do we need all of Parameters/Outputs/Globals to begin with? 
This is an expansion direction, starting out I think we're only focusing on Parameters

Reply:
It's a schema, so let's lay it out now.